packages/core/src/utils.ts

```diff
@@ -31,23 +36,31 @@ export const attachIfSubscribed = (
 };
 
 /**
- * Check if the participant track is pinned.
+ * Check if the `TrackBundle` is pinned.
  */
-export function isParticipantTrackPinned(
-  trackParticipantPair: TrackParticipantPair,
+export function isTrackBundlePinned(
+  trackBundle: TrackBundleWithPlaceholder,
   pinState: PinState | undefined,
 ): boolean {
   if (pinState === undefined) {
     return false;
   }
-
-  const { track, participant } = trackParticipantPair;
-
-  return pinState.some(
-    ({ track: pinnedTrack, participant: pinnedParticipant }) =>
-      pinnedTrack.trackSid === track.trackSid &&
-      pinnedParticipant.identity === participant.identity,
-  );
+  if (isTrackBundle(trackBundle)) {
+    return pinState.some(
+      (pinnedTrackBundle) =>
+        pinnedTrackBundle.participant.identity === trackBundle.participant.identity &&
+        pinnedTrackBundle.source === trackBundle.source &&
+        pinnedTrackBundle.publication.trackSid === trackBundle.publication.trackSid,
```

Comment:
do both conditions actually have to be true here? both source and sid have to match?
If we're checking for SID I would expect that to be enough.